### PR TITLE
Added sse_cli log command

### PR DESCRIPTION
**Pull request by jacobythwaites**
You can now type `sparkl log` to dump the OTP log to console.

### CODE REVIEW

File: sparkl_cli/cmd_log.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    if node is None:
+        node = connection["node"]
+
+    response = sync_request(
+        args, "GET", "sse_log/log",
+        params={
+            "node": node,
+            "type": args.type
+        })
+
```
Comment (miklosduma) [inline sparkl_cli/cmd_log.py]: Me thinks all the other commands return something to the main function, i.e., a generator, or something and leave the "printing" to the main function

File: sparkl_cli/cmd_log.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+def command(args):
+    """
+    Delivers the log data.
+    """
+    connection = get_connection(args)
+    node = args.node
+    if node is None:
+        node = connection["node"]
```
[the remark this answers is not in the document]
Comment (jacobythwaites) [reply sparkl_cli/cmd_log.py]: Nice idea

File: sparkl_cli/cmd_log.py (hunk 1 above)
Comment (jacobythwaites) [reply sparkl_cli/cmd_log.py]: Agree

File: sparkl_cli/cmd_log.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    if args.type not in VALID_REPORTS:
```
Comment (jacobythwaites) [inline sparkl_cli/cmd_log.py]: Nice

File: .travis.yml [hunk 4]
```diff
@@ -10,8 +10,14 @@ before_install:
...
   - python -m sparkl_cli connect http://localhost:8000
   - echo password | python -m sparkl_cli login -r test@localhost password
-  - TEST_SSE=http://localhost:8000 TEST_USER=test@localhost TEST_PASS=password make test
+  - export TEST_SSE=http://localhost:8000
+  - export TEST_USER=test@localhost
+  - export TEST_PASS=password
+  - export TEST_CERT=cert.pem
+  - export TEST_SSE_HTTPS=https://localhost:9000
+  - curl -o ${TEST_CERT} ${TEST_SSE}/sse_cfg/claim -d email=owner@localhost -d password=postoffice
+  - make test
```
Comment (lucafavatellawork) [inline .travis.yml]: I managed to see these passing locally using the amended scripts in PRs #22 and #23, plus work-in-progress scripts for building Docker image from sse_core.

File: sparkl_cli/cmd_log.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
+"""
```
Comment (lucafavatellawork) [inline sparkl_cli/cmd_log.py]: I did not manage to see the `log` by running a local node. I used sse_core repo containing commit b1bcca2, with OTYP 22, with a Dialyzer fix and a patch for supporting OTP 22.

I started the node with `make node`. In other terminal I did:
```
$ python -m sparkl_cli close --all
closed
count:	1
$ python -m sparkl_cli connect "http://localhost:8000"
pong
auth_flows:	sparkl
branch:	HEAD
kernel_machine:	x86_64
kernel_name:	Darwin
kernel_release:	19.0.0
node:	sse1@C02Z30NAL413.local
version:	v1.3.4-18-g31b7
$ python -m sparkl_cli login -r "test@localhost" "password"
user
administrator:	False
id:	G-OCE-1D54-Z
name:	test@localhost
provider:	sparkl
$ python -m sparkl_cli log
exception: {"attr":{"reason":"bad_request"},"tag":"error"}
$ python -m sparkl_cli log
exception: {"attr":{"reason":"bad_request"},"tag":"error"}
$ python -m sparkl_cli 
usage: sparkl_cli [-h] [-v] [-a ALIAS] [-s SESSION] [-t TIMEOUT]
                  {active,call,cd,close,connect,elastic,listen,log,login,logout,ls,mkdir,node,object,put,render,rm,service,session,source,start,stop,tree,undo,user,vars}
                  ...
```

File: .travis.yml [hunk 6]
```diff
@@ -10,8 +10,14 @@ before_install:
...
   - sleep 5
   - python -m sparkl_cli connect http://localhost:8000
   - echo password | python -m sparkl_cli login -r test@localhost password
-  - TEST_SSE=http://localhost:8000 TEST_USER=test@localhost TEST_PASS=password make test
+  - export TEST_SSE=http://localhost:8000
+  - export TEST_USER=test@localhost
+  - export TEST_PASS=password
+  - export TEST_CERT=cert.pem
+  - export TEST_SSE_HTTPS=https://localhost:9000
+  - curl -o ${TEST_CERT} ${TEST_SSE}/sse_cfg/claim -d email=owner@localhost -d password=postoffice
```
Comment (lucafavatellawork) [inline .travis.yml]: Would it be possible to move this step inside the test setup?

File: sparkl_cli/test/test_log.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        logs = sparkl("log", alias=ALIAS)
+
+        first_line = next(logs)
+        assert self.node in first_line
+
+    @pytest.mark.parametrize("report_type", VALID_REPORTS)
+    def test_report_types(self, report_type):
+        logs = sparkl("log", alias=ALIAS, type=report_type)
+        first_line = next(logs)
+        assert self.node in first_line
```
Comment (lucafavatellawork) [inline sparkl_cli/test/test_log.py]: At face value, the tests seem correct. I did not dump what the actual line.

Do these tests have some sort of artefacts that can be inspected for checking the validity of the test?

File: sparkl_cli/cmd_log.py (hunk 5 above)
Comment (jacobythwaites) [reply sparkl_cli/cmd_log.py]: The test@localhost user is not admin, and therefore can't see logs. This is correct behaviour.

File: .travis.yml (hunk 6 above)
Comment (miklosduma) [reply .travis.yml]: Once I'm provided with a way to delete users I'll rework the test setup/teardown so we won't need to handle the setup in three different places (Tests, Makefile, Travis).

File: sparkl_cli/test/test_log.py (hunk 7 above)
Comment (miklosduma) [reply sparkl_cli/test/test_log.py]: If you mean whether the logs are written to some file, no, they are not. If the tests are run locally, one might have a look at the logs connecting to the SPARKL instance the tests use.

**Pull request merged**